pkgs/development/libraries/fftw/default.nix

```diff
@@ -40,22 +37,24 @@ stdenv.mkDerivation (finalAttrs: {
     llvmPackages.openmp
   ] ++ lib.optional enableMpi mpi;
 
-  configureFlags =
-    [ "--enable-shared"
-      "--enable-threads"
-    ]
-    ++ lib.optional (precision != "double") "--enable-${precision}"
-    # all x86_64 have sse2
-    # however, not all float sizes fit
-    ++ lib.optional (stdenv.isx86_64 && (precision == "single" || precision == "double") )  "--enable-sse2"
-    ++ lib.optional enableAvx "--enable-avx"
-    ++ lib.optional enableAvx2 "--enable-avx2"
-    ++ lib.optional enableAvx512 "--enable-avx512"
-    ++ lib.optional enableFma "--enable-fma"
-    ++ [ "--enable-openmp" ]
-    ++ lib.optional enableMpi "--enable-mpi"
-    # doc generation causes Fortran wrapper generation which hard-codes gcc
-    ++ lib.optional (!withDoc) "--disable-doc";
+  configureFlags = [
+    "--enable-shared"
+    "--enable-threads"
+    "--enable-openmp"
+  ]
+
+  ++ lib.optional (precision != "double") "--enable-${precision}"
+  # all x86_64 have sse2
+  # however, not all float sizes fit
+  ++ lib.optional (stdenv.isx86_64 && (precision == "single"))
+    "--enable-sse"
```

Comment:
Ale you sure this is even beneficial?  IIRC `isx86_64` assumes SSE2 already.

Reply:
That's correct, my bad :joy:

Reply:
Can you comment on why you marked this as resolved?  (I'm curious)

Reply:
From my understanding, when fftw is configured with --enable-sse2 it uses sse2 specific code paths.
When it isn't, it uses generic code paths that are built with sse2 instructions but have lower performance.
And in general, SIMD extensions have stricter requirements due to [alignment](https://www.fftw.org/fftw3_doc/SIMD-alignment-and-fftw_005fmalloc.html), so there is definitely a difference.

Reply:
I meant the `--enable-sse` part conditioned on `isx86_64`.

Reply:
> I meant the `--enable-sse` part conditioned on `isx86_64`.

I have removed it